### PR TITLE
Private mode: peer filtering docs

### PR DESCRIPTION
**Pull request by charithabandi**

### CODE REVIEW

File: docs/node/_category_.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "label": "Node",
-    "position": 7,
+    "position": 8,
```
Comment (brennanjl) [inline docs/node/_category_.json:3]: Should still be 7

File: docs/node/private-networks.mdx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+When onboarding a new validator node to the network, the node should first be [added as a read node](#onboarding-a-read-node) to the network. Once the node is part of the network, the node can then request the network to become a validator.
+
+The entire process is as follows:
+- Joiner node is first onboarded as a read node to the network. Refer to the [Onboarding a Read Node](#onboarding-a-read-node) section for more information.
+- Joiner node then requests the network to become a validator by sending a join request to the existing validators. Refer to the [Validator Guide](/docs/node/tutorial#upgrading-the-second-node-to-a-validator) for more information on issuing and approving validator join requests.
+- Existing validators will then approve the join request and adds the joiner node to their whitelisted peers.
+- Once the validator join request is approved, all the nodes would have the joiner node in their whitelist 
+- If the validator join request is rejected, the joiner node will be removed from the whitelist of all the validators that approved the join request.
+
```
Comment (KwilLuke) [inline docs/node/private-networks.mdx:88]: IMO, you can remove all of these steps and just linke to the `/docs/node/tutorial#upgrading-the-second-node-to-a-validator` section.

That also decreases the docs maintenance burden if/when the validator join process changes.

File: docs/node/private-networks.mdx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+:::note
+Whenever a validator is demoted, it is automatically removed from the whitelist of all the nodes in the network. The validator will need to be added back to the whitelist if they want to connect to the network as a read node.
```
Comment (KwilLuke) [inline docs/node/private-networks.mdx:90]: This note is good!

**Review by KwilLuke — CHANGES_REQUESTED** [review]
Good job! A couple of changes, nits, and comments.

Based on your response to a few questions, I may have additional thoughts on how to present the `Onboard Read Node` tutorial.

File: docs/node/private-networks.mdx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+Private mode can also be enabled by setting the `--chain.p2p.private-mode` flag or `KWILD_CHAIN_P2P_PRIVATE_MODE` env variable when starting the node as shown below:
+
+```bash
+kwild -r /path/to/root/dir --chain.p2p.private-mode
+```
+
+
+## Whitelist Peers
+
+When running a Kwild node in private mode, it can be configured to only accept connections from a list of known peers. `whitelist_peers` field should be a comma-separated list of node IDs (ed25519 address).
```
[the remark this answers is not in the document]
Comment (charithabandi) [reply docs/node/private-networks.mdx:32]: It's not a public key, its a list of nodeIDs.

File: docs/node/private-networks.mdx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+:::note
+In Private mode, to establish a connection with the other node, both the nodes should add each other's node ID to the whitelist. If one of the nodes doesn't add the other node to it's whitelist, the connection will not be established between the nodes.
+:::
+
+## Onboarding a Read Node
+
+When onboarding a new read node to the network, the node should be added to the whitelist of at least one of the existing nodes the node wants to connect to. The existing nodes Node ID can be used as a seed node or persistent peer for the new node.
+
+For example, to onboard a new node with the Node ID `new-node-id` to an existing node with the Node ID `existing-node-id`, follow the steps below:
```
[the remark this answers is not in the document]
Comment (charithabandi) [reply docs/node/private-networks.mdx:58]: no, because persistent peers and seed nodes are whitelisted by default

File: docs/node/private-networks.mdx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+Kwild nodes can be run in a private mode, where the node will only accept connections from a list of known peers. This is useful when you want to restrict access to your network to a specific set of nodes.
+
+Private mode can be enabled by setting the `private_mode` field in the `chain.p2p` section of the `config.toml` as shown below:
+
+```toml
+[chain.p2p]
+private_mode = true
+```
+
```
[the remark this answers is not in the document]
Comment (charithabandi) [reply docs/node/private-networks.mdx:22]: kwil-admin doesn't set private_mode. If u r referring to whitelist peers, we mention it below

[the remark this answers is not in the document]
Comment (KwilLuke) [reply docs/node/private-networks.mdx:22]: We can configure other config.toml properties with the `kwil-admin`. I feel like you should be able to do:

`kwil-admin setup peer --chain.p2p.private_mode`

If we can't, I we should add that.

File: docs/node/private-networks.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (KwilLuke) [reply docs/node/private-networks.mdx:59]: Got it - this isn't entirely clear in the `note` above. I'll propose a change.

File: docs/node/private-networks.mdx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+kwild -r /path/to/root/dir --chain.p2p.private-mode --chain.p2p.whitelist-peers="b624b3395f22f8cb57cf86d58cbcd6b355d12e18,7b4b3b3b5f22f8cb57cf86d58cbcd6b355d12e18"
+```
+
+The current set of validators, persistent peers and seed nodes are automatically added to the whitelist. Therefore, the `whitelist_peers` field is used to restrict which non-validating read-only nodes can sync data from your node's P2P endpoint.
+
+The whitelist can also be updated dynamically using the `kwil-admin peers` commands. Refer to the [Kwil Admin Peers Subcommand](/docs/ref/kwil-admin/peers) documentation for more information.
+
+:::note
+In Private mode, to establish a connection with the other node, both nodes should add each other's node ID to the whitelist. If one of the nodes doesn't add the other node to it's whitelist, the connection will not be established, unless it is a seed node or a persistent peer or a current validator.
+:::
```
Comment (KwilLuke) [inline docs/node/private-networks.mdx:52]:
```suggestion
Seed nodes and persistent peers are automatically whitelisted. In Private mode, if the existing node is not a persistent peer or seed node, the connecting node should also add the existing node's ID to its whitelist. 
:::
```

I think something like this, just to make the point immediately that mutual whitelisting is only required if the connected node is not a persistent peer or a seed node.

Comment (KwilLuke) [reply docs/node/private-networks.mdx:52]: Wait so thinking about this practically, when would a connecting node need to add an existing node to its whitelist? How else can it connect without persistent peer or seed nodes?

File: docs/node/private-networks.mdx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    ```toml
+    [chain.p2p]
+    persistent_peers = "existing-node-id@localhost:26656"
+    ```
+## Onboarding a Validator Node
+
+When onboarding a new validator node to the network, the node should first be [added as a read node](#onboarding-a-read-node) to the network. Once the node is part of the network, the node can then [request the network to become a validator](/docs/node/tutorial#upgrading-the-second-node-to-a-validator). 
+
+Existing validators adds the joiner node to their whitelist when they approve the validator join request and if the request is rejected (expired), the joiner node will be removed from the whitelist of all the validators that approved the join request.
+If the validator join request is approved, the new validator node is automatically added to the whitelist of all the nodes in the network.
```
Comment (KwilLuke) [inline docs/node/private-networks.mdx:84]: Typo and clarity:

```suggestion
Existing validators add the joiner node to their whitelist when they approve the validator join request. If the request is rejected (expired), the joiner node will be removed from the whitelist of all the validators that previously approved the request.
If the validator join request reaches the 2/3 vote threshold, the new validator node is automatically added to the whitelist of all the nodes in the network.
```

**Review by KwilLuke — COMMENTED** [review]
Few changes to add clarity to the tutorial.

File: docs/node/private-networks.mdx (hunk 7 above)
Comment (charithabandi) [reply docs/node/private-networks.mdx:52]: none of the persistent peers or seed nodes needs to be whitelisted, but other nodes which are not listed as seeds or persistent peers need to be whitelisted for establishing a mesh connectivity. 

consider 3 nodes: n1 <-> n2, n3(new node)
 suppose n3 is joining the network through n2, here n3 is indirectly connected to n1. So if n1 receives any packet, it broadcasts it to n2 which eventually broadcasts to n3 (takes 2 hops for the packets to reach n3). If n2 goes down, the network breaks. So its good to have 1-1 connectivity with a lot more nodes.  In that case u would add those to your whitelist. 

Cometbft also has a limit on number of persistent peers u can have for better performance, so rest of the nodes u r still connected to, but cometbft doesnt ensure persistence of connection with these nodes.

File: docs/node/private-networks.mdx (hunk 8 above)
Comment (charithabandi) [reply docs/node/private-networks.mdx:84]: 2/3rd threshold is when the validator gets approved, I can just rephrase it to say that explicilty

File: docs/node/private-networks.mdx (hunk 7 above)
Comment (KwilLuke) [reply docs/node/private-networks.mdx:52]: Got it, that makes sense. Thank you for explaining!

I would propose changing it to something like this then:

```suggestion
Seed nodes and persistent peers are automatically whitelisted. In order for a new node to maintain a connection with an existing node that is not a seed or persistent peer, the existing node must also add the new node to its whitelist.
:::
```

File: docs/node/private-networks.mdx (hunk 6 above)
[the remark this answers is not in the document]
Comment (charithabandi) [reply docs/node/private-networks.mdx:22]: There are bunch of changes coming to kwil-admin setup, will add along with that.

**Comment by KwilLuke** [comment]
Changes look good to me! Once the `kwil-admin` comment is resolved, we can merge this.

**Review by KwilLuke — APPROVED** [review]
lgtm
